wordpress-mobile/WordPress-Android
# Localization support for about-automattic-android library

This PR adds support for importing string from the `Automattic/about-automattic-android` library. `release-toolkit` didn't have support for downloading private repository files, so it was added in https://github.com/wordpress-mobile/release-toolkit/pull/321. The `fastlane-plugin-wpmreleasetoolkit` gem is currently pointing to this PR and has to be updated with a proper release before merging.

**Merge instructions**
1. [x] Review & merge 23-gh-Automattic/about-automattic-android. This PR renames the generic `app_name` string resource which would otherwise override the name of the app.
2. [x] Create a new tag/release for `Automattic/about-automattic-android`, using `0.0.2` as the tag name
3. [x] Once Buildkite publishes the library ^, update `ext.aboutAutomatticVersion` in `build.gradle` to `0.0.2`
4. [x] Review & merge https://github.com/wordpress-mobile/release-toolkit/pull/321 which adds support for downloading files for private GitHub repositories
5. [x] Create a new `release-toolkit` release
6. [x] Update `fastlane-plugin-wpmreleasetoolkit` in `fastlane/Pluginfile` for the new release. Assuming we create a minor release, reverting the change in that file should be enough.
7. [x] Review & test this PR and finally merge it.

**To test**
* Create a new branch from this PR (not strictly necessary)
* Comment out [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/about-library-localization/fastlane/Fastfile#L814) so once `localize_binary_deps` is called, it won't create a commit
* Run `bundle exec fastlane localize_binary_deps`
* Verify that there has been some changes to the Gutenberg related strings in `WordPress/src/main/res/values/strings.xml`
* Verify that the strings in `library/src/main/res/values/strings.xml` of `Automattic/about-automattic-android` library has been added at the end of `WordPress/src/main/res/values/strings.xml`
* Verify that `app_name` string in `WordPress/src/main/res/values/strings.xml` is not changed

## Regression Notes
1. Potential unintended areas of impact
Strings for `wordpress-mobile/gutenberg-mobile`, `wordpress-mobile/WordPress-Login-Flow-Android` & `Automattic/stories-android` might not be downloaded during code freeze. The possibility is there because https://github.com/wordpress-mobile/release-toolkit/pull/321 adds an extra step to find the download url of the resource by making a GitHub API request.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I've manually tested the `localize_binary_deps` lane and added test instructions for it to the PR.

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**
